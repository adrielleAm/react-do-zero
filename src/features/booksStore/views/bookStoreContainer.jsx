import React from 'react';
import BookStorePage from './bookStorePage';
import { connect } from 'react-redux';

const BookStoreContainer = props => {
  return <BookStorePage books={props.books} />;
};

const mapStateToProps = ({ bookStoreReducer }) => {
  return {
    books: bookStoreReducer.books
  };
};

export default connect(mapStateToProps)(BookStoreContainer);
