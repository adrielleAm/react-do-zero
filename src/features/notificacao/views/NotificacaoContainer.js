import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Notificacao from './notificacao';
import { notificacaoActions } from '../redux';

export class NotificacaoContainer extends Component {
  state = {
    exibeNotificacao: false
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.reduxProps.exibeNotificacao !== prevState.exibeNotificacao) {
      return { exibeNotificacao: nextProps.reduxProps.exibeNotificacao };
    } else return prevState;
  }

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.props.reduxProps.exibeNotificacao !==
      nextProps.reduxProps.exibeNotificacao
    );
  }

  render() {
    return (
      <React.Fragment>
        <Notificacao
          {...{
            ...this.props.reduxProps,
            ocultaNotificacao: this.props.ocultaNotificacao
          }}
        />
      </React.Fragment>
    );
  }
}
NotificacaoContainer.defaultProps = {
  reduxProps: {}
}

NotificacaoContainer.prototypes = {
  reduxProps: PropTypes.object.isRequired,
  ocultaNotificacao: PropTypes.func.isRequired
};

const mapStateToProps = ({ notificacao }) => {
  return {
    reduxProps: {
      mensagem: notificacao.mensagem,
      exibeNotificacao: notificacao.exibeNotificacao,
      variant: notificacao.variant
    }
  };
};
const mapDispatchToProps = {
  ...notificacaoActions
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotificacaoContainer);
