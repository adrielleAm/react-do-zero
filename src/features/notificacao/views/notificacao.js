import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CloseIcon from '@material-ui/icons/Close';
import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';
import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import { withStyles } from '@material-ui/core/styles';

export const notificacao = props => {
  const { classes, className, mensagem, variant, exibeNotificacao, ocultaNotificacao } = props;
  return (
    <Snackbar
      id="snackBarNotificacao"
      anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
      open={exibeNotificacao}
      autoHideDuration={4000}
      onClose={ocultaNotificacao}
      style={{ display: mensagem.length ? 'block' : 'none' }}
    >
      {mensagem.length ? (
        <SnackbarContent
          id="snackBackContent"
          className={classNames(classes[variant], className)}
          aria-describedby="client-snackbar"
          message={mensagem}
          action={[
            <IconButton
              id="iconOcultarNotificacao"
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={ocultaNotificacao}
            >
              <CloseIcon />
            </IconButton>
          ]}
        />
      ) : null}
    </Snackbar>
  );
};

notificacao.defaultProps = {
  variante: 'success',
  mensagem: ''
};

notificacao.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  mensagem: PropTypes.node || PropTypes.string,
  onClose: PropTypes.func,
  variant: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  ocultaNotificacao: PropTypes.func
};

const styles = theme => ({
  success: {
    backgroundColor: green[600]
  },
  error: {
    backgroundColor: theme.palette.error.dark
  },
  info: {
    backgroundColor: theme.palette.primary.dark
  },
  warning: {
    backgroundColor: amber[700]
  },
  icon: {
    fontSize: 20
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing.unit
  },
  message: {
    display: 'flex',
    alignItems: 'center'
  }
});

export default withStyles(styles)(notificacao);
