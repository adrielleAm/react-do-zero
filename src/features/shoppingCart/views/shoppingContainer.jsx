import React from 'react';
import ShoppingCartPage from './shoppingCartPage';
import { connect } from 'react-redux';
const shoppingContainer = props => {
  console.log('props :', props);
  return <ShoppingCartPage compraEfetuada={props.compraEfetuada} />;
};
const mapStateToProps = ({ shoppingCart }) => {
  return {
    compraEfetuada: shoppingCart.compraEfetuada
  };
};

export default connect(mapStateToProps)(shoppingContainer);
