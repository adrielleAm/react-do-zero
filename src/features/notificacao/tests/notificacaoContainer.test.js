import React from 'react';
import { shallow } from 'enzyme';
import { NotificacaoContainer } from '../views/NotificacaoContainer';
import  Notificacao  from '../views/notificacao';

describe('Testa <NotificacaoContainer />', () => {
  let component;
  beforeEach(() => {
    component = shallow(<NotificacaoContainer reduxProps={{}}/>);
  });
  it('Deve renderizar <NotificacaoContainer /> ', () => {
    expect(component).toMatchSnapshot();
  });
  it('Deve renderizar <Notificacao /> ', () => {
    expect(component.find(Notificacao).exists()).toEqual(true);
  });

  it('Deve setar props exibeNotificacao para true e state exibeNotificacao deve ser true ', () => {
    component.setProps({
      reduxProps: { exibeNotificacao: true }
    });
    expect(component.state().exibeNotificacao).toEqual(true);
  });

  it('Deve setar props exibeNotificacao para false e state exibeNotificacao deve ser false ', () => {
    component.setProps({
      reduxProps: { exibeNotificacao: false }
    });
    expect(component.state().exibeNotificacao).toEqual(false);
  });
});
