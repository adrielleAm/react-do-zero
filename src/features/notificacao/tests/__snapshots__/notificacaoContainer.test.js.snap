// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa <NotificacaoContainer /> Deve renderizar <NotificacaoContainer />  1`] = `
<Fragment>
  <WithStyles(notificacao) />
</Fragment>
`;
